src/util/string/hash.js

```diff
@@ -0,0 +1,12 @@
+define(function() {
+
+// Based on http://stackoverflow.com/questions/7616461/generate-a-hash-from-string-in-javascript-jquery
+return function( str ) {
+	return [].reduce.call( str, function( hash, i ) {
```

Comment:
The faster this code, the better. I'm curious how faster your suggested code is compared to this one. The difference is using `for` instead of the ES5 `reduce`.